docker/Dockerfile

```diff
@@ -4,12 +4,25 @@
 FROM python:3.9-slim-bullseye
 ARG SDK_REQUIREMENT
 
+# Set by BuildKit
+ARG TARGETARCH
+
+# If statement installs any additional dependencies needed for building for ARM
+# (development using an M1 is fun :') )
 RUN <<EOF
-apt update --yes
-apt upgrade --yes
-apt install --yes --no-install-recommends git openssh-client
-python -m pip install -U pip wheel
-python -m pip install "${SDK_REQUIREMENT}"
+set -ex
+apt-get update --yes
+apt-get upgrade --yes
+apt-get install --yes --no-install-recommends git openssh-client
+
+if [ "$TARGETARCH" = "arm64" ]; then
+  apt-get install --yes --no-install-recommends gcc libc-dev
+fi
+
+rm -rf /var/lib/apt/lists/*
```

Comment:
**suggestion**: A few more cleanup commands (Doubt they'll save much more, but maybe just for completeness)
```suggestion
apt-get autoremove --yes
apt-get clean
rm -rf /var/lib/apt/lists/*
```

Reply:
> Official Debian and Ubuntu images [automatically run apt-get clean](https://github.com/moby/moby/blob/03e2923e42446dbb830c654d0eec323a0b4ef02a/contrib/mkimage/debootstrap#L82-L105), so explicit invocation is not required.

https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#apt-get

I don't think we need to run autoremove either, because we never uninstall anything and autoremove just removes unused dependencies, but happy to add it if you think it's needed.